runtime/lua/vim/diagnostic.lua

```diff
@@ -2150,43 +2277,69 @@ function M.open_float(opts, ...)
   end
 
   local scope = ({ l = 'line', c = 'cursor', b = 'buffer' })[opts.scope] or opts.scope or 'line'
-  local lnum, col --- @type integer, integer
-  local opts_pos = opts.pos
+  local extmarks_start = { 0, 0 }
+  local extmarks_end = { -1, -1 }
   if scope == 'line' or scope == 'cursor' then
+    local opts_pos = opts.pos
+
     if not opts_pos then
       local pos = api.nvim_win_get_cursor(0)
-      lnum = pos[1] - 1
-      col = pos[2]
+      extmarks_start[1] = pos[1] - 1
+
+      if scope == 'cursor' then
+        extmarks_start[2] = pos[2]
+      end
     elseif type(opts_pos) == 'number' then
-      lnum = opts_pos
+      extmarks_start[1] = opts_pos
     elseif type(opts_pos) == 'table' then
-      lnum, col = opts_pos[1], opts_pos[2]
+      extmarks_start[1] = opts_pos[1]
+
+      if scope == 'cursor' then
+        extmarks_start[2] = opts_pos[2]
+      end
     else
       error("Invalid value for option 'pos'")
     end
+
+    extmarks_end[1] = extmarks_start[1]
+
+    if scope == 'cursor' then
+      extmarks_end[2] = extmarks_start[2]
+    end
   elseif scope ~= 'buffer' then
     error("Invalid value for option 'scope'")
   end
 
-  local diagnostics = get_diagnostics(bufnr, opts --[[@as vim.diagnostic.GetOpts]], true)
+  local namespaces = {} ---@type integer[]
+  if type(opts.namespace) == 'number' then
+    namespaces[1] = opts.namespace
+  else
+    for namespace in pairs(diagnostic_cache[bufnr]) do
+      table.insert(namespaces, namespace)
+    end
+  end
```

Comment:
should `type(opts.namespace) == 'table'` case be supported?

Neither types nor docs seem to allow it, but implicitly it has been supported since
`get_diagnostics` does

Reply:
Opened https://github.com/neovim/neovim/pull/34073 for this.

Reply:
Added it here 3c325a08bdafaf02e7b61fd64634d5f1dcd9842f
Only implementation change.